docs/source/deploy_chaincode.md

```diff
@@ -0,0 +1,348 @@
+- [Step two: Install the chaincode package](#install-the-chaincode-package)
+- [Step three: Approve a chaincode definition](#approve-a-chaincode-definition)
+- [Step four: Committing the chaincode definition to the channel](#committing-the-chaincode-definition-to-the-channel)
+
+
+## Setup Logspout (optional)
+
+This step is not required, but is extremely useful for troubleshooting chaincode. To monitor the logs of the smart contract, an administrator can view the aggregated output from a set of Docker containers using the `logspout` [tool](https://logdna.com/what-is-logspout/). The tool collects the output streams from different Docker containers into one place, making it easy to see what's happening from a single window. This can help administrators debug problems when they install smart contracts or developers when they invoke smart contracts. Because some containers are created purely for the purposes of starting a smart contract and only exist for a short time, it his helpful to collect all of the logs from your network.
+
+A script to install and configure Logspout, `monitordocker.sh`, is already included in the Commercial Paper tutorial. We will use the same script in this tutorial as well. The Logspout tool will continuously stream logs to your terminal, so you will need to use a new terminal window. Open a new terminal and navigate to the `test-network` directory.
+```
+cd fabric-samples/test-network
+```
+
+You can run the `monitordocker.sh` script from any directory. For ease of use, we will copy the the `monitordocker.sh` script from the `commercial-paper` sample to your working directory
+```
+cp ../commercial-paper/organization/digibank/configuration/cli/monitordocker.sh .
+# if you're not sure where it is
+find . -name monitordocker.sh
+```
+
+You can then start Logspout by running the following command:
+```
+./monitordocker.sh net_test
+```
+You should see output similar to the following:
+```
+Starting monitoring on all containers on the network net_basic
+Unable to find image 'gliderlabs/logspout:latest' locally
+latest: Pulling from gliderlabs/logspout
+4fe2ade4980c: Pull complete
+decca452f519: Pull complete
+ad60f6b6c009: Pull complete
+Digest: sha256:374e06b17b004bddc5445525796b5f7adb8234d64c5c5d663095fccafb6e4c26
+Status: Downloaded newer image for gliderlabs/logspout:latest
+1f99d130f15cf01706eda3e1f040496ec885036d485cb6bcc0da4a567ad84361
+
+```
+You will not see any logs at first, but this will change when we deploy our chaincode. It can be helpful to make this window wide and the font small.
+
+## Package the smart contract
+
+We need to package the chaincode before it can be installed on our peers. The steps are different if you want to install a smart contract written in [Golang](#golang), [Java](#java), or [JavaScript](#javascript).
+
+### Golang
+
+Before we package the chaincode, we need to install the chaincode dependences. Navigate to the folder that contains the go version of the Fabcar chaincode.
```

Comment:
Would it make any sense to put these behind a twistie? Or add 3 links under Package the smart contract so they can go directly to the instructions for the language they care about?

Reply:
The link is there: The steps are different if you want to install a smart contract written in [Go](#go), [Java](#java), or [JavaScript](#javascript).